node/common/transport/mqtt/lib/mqtt.d.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { Packet } from 'mqtt';
+import { X509, results } from 'azure-iot-common';
+import MqttReceiver = require('./mqtt_receiver');
+
+declare class Mqtt {
+    // There is currently no way for the parameter to be a whole TypeScript namespace
+    constructor(mqttprovider: any);
+    connect(config: Mqtt.ConnectOptions, done: (err: Error, ack?: Packet) => void): void;
+    disconnect(done: () => void): void;
+    publish(message: { data: { toString(): string }}, done: (err: Error, result?: results.MessageEnqueued) => void): void;
+    getReceiver(done: (err: Error, receiver: MqttReceiver) => void): void;
+}
+
+declare namespace Mqtt {
+    interface ConnectOptions {
```

Comment:
Yup, all device transport configs are the same, and similar to Mqtt base config so I made them extend that in next commit, however bases don't have a common configuration. As for service it doesn't have deviceId altogether and sharedAccessSignature is required (x509 not supported) so it's a seperate base, might be worth merging them when we get support for x509.